sycl/test/basic_tests/get_nonhost_devices.cpp

```diff
@@ -1,6 +1,8 @@
-// RUN: %clangxx -fsycl %s -o %t.out
-// RUN: %t.out
-
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
```

Comment:
using SYCL_DEVICE_TYPE affects only default_selector (see https://github.com/intel/llvm/blob/sycl/sycl/source/device_selector.cpp#L89)

There is no reason for the change.

Reply:
On a machine with just the NVIDIA OpenCL installed the above fails (sadly on an automatic testing machine I do not have debugging access to).

The `XYZ_RUN_PLACEHOLDER` also adds the `SYCL_BE=PI_XYZ` environment variable and replaces the CPU and ACC runs. With just an NVIDIA OpenCL installed a `check-sycl-cuda` test run then avoids OpenCL and passes.

Alternatively, I could mark the test as `REQUIRES: opencl` so it would be ignored in `check-sycl-cuda` test runs.

Reply:
"REQUIRES: opencl" is more clear from my perspective.

Reply:
Thank you, I will go with that.

Reply:
Done. Will be in next push.